crates/core/src/doc/pluck.rs

```diff
@@ -164,8 +165,12 @@ impl Document {
 		}
 		// Remove any omitted fields from output
 		if let Some(v) = stm.omit() {
-			for v in v.iter() {
-				out.del(stk, ctx, opt, v).await?;
+			let fields = exprs_to_fields(stk, ctx, opt, Some(&self.current), v.as_slice())
```

Comment:
This is likely to have a negative impact on performances.  The expression is re-evaluated for each document. Is there a possibility to pre-compile it once per statement?

Reply:
Yes, good point! Looking